### ISSUE TITLE
[$2000] Web - Push notification - Clicking the notification takes you to the incorrect chat first

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10327]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with applausetester@applause.expensifail.com
account in device 1 
3. Login in another device as account B
4. In device 1 go to another chat and minimize the browser
5. Send some messages from device 2 to device 1
6. Click on notification of the new message in device 1 

## Expected Result:
Clicking the notification takes you to the correct chat

## Actual Result:
Clicking the notification takes you to the incorrect chat first

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.88.7

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** applausetester@applause.expensifail.com/1Password

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/183735475-5c128a05-844f-42c5-a2e0-c00cf9ace648.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alex-mechler** [comment]
Is any action needed to get the chats to switch, or does it happen automatically, it just takes a moment?

**Comment by kavimuru** [comment]
It just switches to correct chat after a moment, no action required.

**Comment by alex-mechler** [comment]
Awesome, thanks!

I think this is an underlying performance issue, but still going to send this external to see if there is anything we can do to speed this up while the larger performance updates are happening!

**Comment by kadiealexander** [comment]
https://www.upwork.com/jobs/~0109489e265df8d07e
https://www.upwork.com/ab/applicants/1557956048865767424/job-details

**Comment by Julesssss** [comment]
> I think this is an underlying performance issue, but still going to send this external to see if there is anything we can do to speed this up while the larger performance updates are happening!

Okay. So let's just aim for improvement here. We can close out the issue if no good solution is made.

**Comment by iyinolu** [comment]
Is this Issue still open for contributions?

**Comment by iyinolu** [comment]
Please verify you have provided the correct password for the affected acccount. Thanks

**Comment by iyinolu** [comment]
Issue is not reproducible on MacOS. Push Notification links to the correct chat only [staging env].

**Comment by Julesssss** [comment]
hey @kadiealexander, can we please double the issue amount?

**Comment by kadiealexander** [comment]
Sorry, was flying! Price doubled!

**Comment by kadiealexander** [comment]
Price doubled again!

**Comment by kadiealexander** [comment]
Price doubled again!

**Comment by kadiealexander** [comment]
Reassigning as I'm ooo for the next 9 days.

**Comment by trjExpensify** [comment]
Oh, hm. @Julesssss - weren't you looking at a broader proposal for improving notifications in NewDot. Wouldn't this be included in that initiative?

**Comment by Julesssss** [comment]
Yep, I added this to [the list](https://github.com/Expensify/Expensify/issues/219254) but let's keep it open for external contributors in the meantime.

**Comment by trjExpensify** [comment]
Okay, but don't we run the risk of just another piecemeal fix ere, versus holistically revisiting notifications?

**Comment by Julesssss** [comment]
A fix would not necessarily be incompatible with further improvement, and there is a benefit to immediately fixing this, rather than awaiting a DD. IIRC this used to work...

But I am also happy to put the issue on hold while the project is moved forward.

**Comment by trjExpensify** [comment]
I can't really remember tbh. In the absence of comment linking for the message, I always thought it was spotty. But cool, happy to leave it open for the time being.

**Comment by Julesssss** [comment]
Cool. Let's see what sort of proposals we recieve. I'll bear the wider project in mind when deciding to move forward or not.

**Comment by trjExpensify** [comment]
Waiting on proposals. 

I have a feeling that this is an underlying performance issue where it takes a moment to switch to the correct chat report. I don't think it "navigates to the incorrect chat", but rather it's slow to move from the chat that you're currently on for a few seconds. 

@JmillsExpensify - this is prolly worth adding to the WQ project and I wonder if our optimisations around performance with Margelo will ultimately be how we solve it.

**Comment by trjExpensify** [comment]
^^ Thoughts on the above @JmillsExpensify @Julesssss?

**Comment by Julesssss** [comment]
Yeah perhaps we should put this on hold and deal with it internally at a higher priority

**Comment by trjExpensify** [comment]
I think so, or maybe even close. I'm unable to reproduce it on prod. Things seem speedy, what about you? CC: @twisterdotcom 

![notification web switch](https://user-images.githubusercontent.com/16232057/193033596-9628ca36-aacd-42fb-b1fa-0eddd07c7fb2.gif)

**Comment by trjExpensify** [comment]
Bump, can anyone reproduce this now?

**Comment by Julesssss** [comment]
Sorry, I don't have any time to look at these steps today 😩

**Comment by trjExpensify** [comment]
Cool. It would be good to confirm this one to double or close. @mananjadhav can you try as well?

**Comment by JmillsExpensify** [comment]
(Missed this earlier. Added to the WA Quality project).

**Comment by mananjadhav** [comment]
I did try few notifications and I wasn't able to reproduce this. I'll take another stab at it today.

**Comment by Julesssss** [comment]
> Bump, can anyone reproduce this now?

@trjExpensify I'm unable to reproduce on web or Android (only platforms I tested)

**Comment by Julesssss** [comment]
Let's close.

**Comment by Julesssss** [comment]
Could be a side effect of the performance improvements we've been making. The app opened immediately for me which is unusually rare for me on Android.

**Comment by Julesssss** [comment]
Here's the Android screen capture I forgot to attach: 


https://user-images.githubusercontent.com/10736861/194536240-2e9d8e8a-479f-4025-ba51-f0156e8685ea.mp4

**Comment by trjExpensify** [comment]
Cool, sounds good! Thanks for confirming.
